protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt

```diff
@@ -44,28 +48,34 @@ import com.toasttab.protokt.codegen.protoc.Tag
 import com.toasttab.protokt.codegen.template.Message.Message.DeserializerInfo
 import com.toasttab.protokt.codegen.template.Message.Message.DeserializerInfo.Assignment
 import com.toasttab.protokt.codegen.template.Message.Message.PropertyInfo
-import com.toasttab.protokt.codegen.template.Renderers.Deserialize
-import com.toasttab.protokt.codegen.template.Renderers.Deserialize.Options
-import com.toasttab.protokt.codegen.template.Renderers.Read
 import com.toasttab.protokt.rt.KtDeserializer
 import com.toasttab.protokt.rt.KtMessageDeserializer
-import com.toasttab.protokt.codegen.template.Oneof as OneofTemplate
+import com.toasttab.protokt.rt.UnknownFieldSet
 
 internal class DeserializerAnnotator
 private constructor(
     private val msg: Message,
     private val ctx: Context
 ) {
+    /** Creates a {@see KtDeserializer<T>} companion object, where T is the Kotlin type for this Message.
+     *
+     * Contains functions:
+     *  - deserialize(deserializer : KTMessageDeserializer) : T
+     *  - invoke(dsl: ApiDsl.() -> Unit): Api
+     *
+     * */
     private fun annotateDeserializer(): TypeSpec {
         val deserializerInfo = annotateDeserializerOld()
         val properties = annotateProperties(msg, ctx)
 
         return TypeSpec.companionObjectBuilder("Deserializer")
+            // KtDeserializer<MessageType>
             .addSuperinterface(
                 KtDeserializer::class
                     .asTypeName()
                     .parameterizedBy(msg.typeName)
             )
+            // (MessageTypeDsl.() -> Unit) -> Message
```

Comment:
Hmm... not sold on the necessity of these comments. Perhaps extracting to a named function would be good?